test/utils/deploy/deploy.go

```diff
@@ -37,6 +37,7 @@ import (
 	"github.com/openshift-kni/numaresources-operator/controllers"
 	nropmcp "github.com/openshift-kni/numaresources-operator/internal/machineconfigpools"
 	"github.com/openshift-kni/numaresources-operator/pkg/status"
+	"github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests"
```

Comment:
please no: code in `test/e2e/serial/tests` may depend on these utilities but not the other way around

Reply:
Done. I moved the MCP wait logic to `test/utils/deploy/deploy.go` and 
now `github.com/openshift-kni/numaresources-operator/test/e2e/serial/tests` consumes it